python/task_manager/background_task_manager.py

```diff
@@ -20,6 +20,11 @@
 from .worker_thread import WorkerThread
 from .results_poller import ResultsDispatcher
 
+# Set to True to enable extensive debug logging.
+# Useful for debugging concurrency issues.
+# Do not release with this setting set to True!
+ENABLE_DETAILED_DEBUG = False
```

Comment:
I like environment variables for these sort of things. This way you can never commit the flag with True.